### PR TITLE
Create action-security.yml

### PR DESCRIPTION
**Pull request by arjundashrath**
@varunsh-coder I had some trouble deploying the action on my test repository as it was being deemed invalid in the builds, hence the harden runner link does not really show anything for the outbound endpoints, any suggestions? I have updated the github token permissions

### CODE REVIEW

**Comment by arjundashrath** [comment]
closes #185

File: knowledge-base/azure/k8s-deploy/action-security.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: 'Deploy to Kubernetes cluster'
+github-token:
+  action-input:
+    input: token
+    is-default: true
+  permissions:
+    contents: read
+    contents-reason: to query workflow path
+    issues: write
+    issues-reason: to set issues as idle
```
Comment (varunsh-coder) [inline knowledge-base/azure/k8s-deploy/action-security.yml:10]: @arjundashrath can you please add a link to the reason, where it has a reason the contents-read and others are needed. That makes it easy for me to verify this. You can add it as a comment. See [this](https://github.com/step-security/secure-workflows/blob/8e3478a5c48e4e82ab08272884d0051785095c71/knowledge-base/Azure/static-web-apps-deploy/action-security.yml#L4) for example.

Comment (arjundashrath) [reply knowledge-base/azure/k8s-deploy/action-security.yml:10]: https://github.com/Azure/k8s-deploy/blob/ac49626466364585521685ccaad17e499a8263a8/src/utilities/utility.ts#L88

https://github.com/Azure/k8s-deploy/blob/acd12a47055b679be711e4937c41f1755316cfae/.github/workflows/defaultLabels.yml#L20
the workflows contain token for Issue/PR management, I wonder if it takes it from the action input?

File: knowledge-base/azure/k8s-deploy/action-security.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+name: 'Deploy to Kubernetes cluster'
+github-token:
+  action-input:
+    input: token
+    is-default: true
+  permissions:
+    contents: read
```
Comment (varunsh-coder) [inline knowledge-base/azure/k8s-deploy/action-security.yml:7]: I did not see contents: read or issues: write.
This just needs `actions: read` and the link is https://github.com/Azure/k8s-deploy/blob/ac49626466364585521685ccaad17e499a8263a8/src/githubClient.ts#L11

Comment (varunsh-coder) [reply knowledge-base/azure/k8s-deploy/action-security.yml:7]: @arjundashrath this one is not updated.

**Pull request merged**